src/libmongoc/tests/test-mongoc-max-staleness.c

```diff
@@ -57,9 +57,15 @@ test_mongoc_client_max_staleness (void)
    capture_logs (true);
    ASSERT (!test_framework_client_new (
       "mongodb://a/?" MONGOC_URI_MAXSTALENESSSECONDS "=120", NULL));
+   ASSERT_CAPTURED_LOG (MONGOC_URI_MAXSTALENESSSECONDS "=120",
+                        MONGOC_LOG_LEVEL_WARNING,
+                        "Invalid readPreferences");
+
+   capture_logs (true);
```

Comment:
The test was previously using `ASSERT_CAPTURED_LOG` once for two separate calls to `test_framework_client_new`. This ensures each is tested independently, since `capture_logs()` resets the log each time.